hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/resolver/TestNamenodeResolver.java

```diff
@@ -359,11 +359,12 @@ public void testRegistrationNamenodeSelection()
         FederationNamenodeServiceState.ACTIVE);
 
     // 1) ns0:nn0 - Standby (oldest)
-    // 2) ns0:nn1 - Standby (newest)
-    // 3) ns0:nn2 - Standby
-    // Verify the selected entry is the newest standby entry
+    // 2) ns0:nn1 - Standby
+    // 3) ns0:nn2 - Standby (newest)
+    // Verify the selected entry is the oldest standby entry
     assertTrue(namenodeResolver.registerNamenode(createNamenodeReport(
         NAMESERVICES[0], NAMENODES[0], HAServiceState.STANDBY)));
+    Thread.sleep(1500);
```

Comment:
We shouldn't be adding waits like this, can we add a GenericTestUtils#waitFor or similar?

Reply:
This suggestion sounds good, I added GenericTestUtils#atLeastWaitFor method, please help to take a look.